src/plugins/ucc_discovery_plugin.cpp

```diff
@@ -162,6 +190,11 @@ bool UccDiscoveryPlugin::_dictionary_segments_contain_duplicates(const std::shar
       continue;
     }
 
+    if (source_chunk->invalid_row_count() != 0 || source_chunk->is_mutable()) {
+      // The segment might have been modified. Because the modification might consist of deleting a duplicate value,
+      // we can't be sure that the segment still has duplicates using the heuristic employed below.
+      return false;
+    }
```

Comment:
Mutable segments are always unencoded. Thus, the second condition is not necessary here (code will return false anyway).